keyboards/keebio/iris/keymaps/jestes5111/keymap.c

```diff
@@ -0,0 +1,170 @@
+    // ├──────────┼──────────┼──────────┼──────────┼──────────┼──────────┤                              ├──────────┼──────────┼──────────┼──────────┼──────────┼──────────┤
+        KC_NO,     KC_NO,     KC_NO,     KC_NO,     KC_NO,     KC_NO,                                    THUMB,     KC_NO,     KC_NO,     KC_NO,     KC_NO,     KC_NO,
+    // ├──────────┼──────────┼──────────┼──────────┼──────────┼──────────┤                              ├──────────┼──────────┼──────────┼──────────┼──────────┼──────────┤
+        KC_NO,     KC_NO,     KC_NO,     KC_NO,     KC_NO,     KC_NO,                                    KC_NO,     KC_NO,     KC_NO,     KC_NO,     KC_NO,     KC_NO,
+    // ├──────────┼──────────┼──────────┼──────────┼──────────┼──────────┼──────────┐        ┌──────────┼──────────┼──────────┼──────────┼──────────┼──────────┼──────────┤
+        KC_NO,     KC_NO,     KC_NO,     KC_NO,     KC_NO,     KC_NO,     KC_NO,              KC_NO,     KC_NO,     KC_NO,     KC_NO,     KC_NO,     KC_NO,     KC_NO,
+    // └──────────┴──────────┴──────────┴────┬─────┴────┬─────┴────┬─────┴────┬─────┘        └────┬─────┴────┬─────┴────┬─────┴────┬─────┴──────────┴──────────┴──────────┘
+                                            TO(_QWERTY), KC_TRNS,   KC_NO,                         KC_NO,     KC_TRNS,   KC_NO
+    //                                       └──────────┴──────────┴──────────┘                   └──────────┴──────────┴──────────┘
+    ),
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case THUMB:
+            if (record->event.pressed) {
+                SEND_STRING(":disguised_face: :thumbsup:");
+                wait_ms(100);
+                tap_code(KC_ENTER);
+            }
+            return false;
+            break;
+    }
+    return true;
+}
+
+void keyboard_post_init_user(void) {
+    rgblight_enable_noeeprom();
+    rgblight_sethsv_noeeprom(HSV_PURPLE);
+    rgblight_mode_noeeprom(RGBLIGHT_MODE_STATIC_LIGHT);
+}
+
+void eeconfig_init_user(void) {
+    rgblight_enable();
+    rgblight_sethsv(HSV_PURPLE);
+    rgblight_mode(RGBLIGHT_MODE_STATIC_LIGHT);
+}
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+    switch (get_highest_layer(state)) {
+        case _LOWER:
+            rgblight_sethsv_noeeprom(HSV_GREEN);
+            break;
+        case _RAISE:
+            rgblight_sethsv_noeeprom(HSV_RED);
+            break;
+        case _ADJUST:
+            rgblight_sethsv_noeeprom(HSV_WHITE);
+            break;
+        case _FUNC:
+            rgblight_sethsv_noeeprom(HSV_BLUE);
+            break;
+        case _PASTA:
+            rgblight_sethsv_noeeprom(HSV_ORANGE);
+            break;
+        default:
+            rgblight_sethsv_noeeprom(HSV_PURPLE);
+            break;
+    }
+    return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
```

Comment:
FYI: You could use the new [Tri Layer feature](https://docs.qmk.fm/#/feature_tri_layer) for this, but since you are doing other stuff on this function (changing colors) i dont think it is wrong using the "old way"

Btw, wouldn't you need the operations in the reverse order so that the indicators "see" the correct state for `_ADJUST`?
```c
state = update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
switch (get_highest_layer(state)) {
    // ...
}
return state;
 ```

Reply:
I was under the impression that I needed `update_tri_layer_state` and that it wouldn't do it automatically, but it appears that is not the case. Thanks for the suggestion!

As for your second point, it didn't seem to matter in my (rather limited) testing. I was able to use my `_ADJUST` layer with no issues, at least.

Reply:
> I was under the impression that I needed `update_tri_layer_state` and that it wouldn't do it automatically, but it appears that is not the case. Thanks for the suggestion!

It doesnt do it automatically, unless you enable the new feature i linked to.... You need to either enable that, or un-do the removal of `layer_state_set_user`

> As for your second point, it didn't seem to matter in my (rather limited) testing. I was able to use my `_ADJUST` layer with no issues, at least.

I guess the `update_tri_layer_state` call was triggering a second call to `layer_state_set_user` and it worked anyway

Reply:
> It doesnt do it automatically, unless you enable the new feature i linked to.... You need to either enable that, or un-do the removal of update_tri_layer_state

Isn't that what having `TRI_LAYER_ENABLE = yes` in `rules.mk` does? Or did I miss a step here?

Reply:
> Or did I miss a step here?

it was **ME** who missed whichever commit where you added it, it should be good to go indeed 😅